FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs

```diff
@@ -413,6 +414,77 @@ public async Task<UserRecord> GetUserByPhoneNumberAsync(string phoneNumber)
                 .ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Gets the user data corresponding to the specified identifiers.
+        /// <para>
+        /// There are no ordering guarantees; in particular, the nth entry in the users result list
+        /// is not guaranteed to correspond to the nth entry in the input parameters list.
+        /// </para>
+        /// <para>
+        /// A maximum of 100 identifiers may be supplied. If more than 100 identifiers are specified,
+        /// this method throws an <c>ArgumentException</c>.
+        /// </para>
+        /// </summary>
+        /// <param name="identifiers">The identifiers used to indicate which user records should be
+        /// returned. Must have 100 entries or fewer.</param>
+        /// <returns>A {@code Task} that resolves to the corresponding user records.</returns>
+        /// <exception cref="ArgumentException">If any of the identifiers are invalid or if more
+        /// than 100 identifiers are specified.</exception>
+        public async Task<GetUsersResult> GetUsersAsync(
+            IReadOnlyCollection<UserIdentifier> identifiers)
+        {
+            return await this.GetUsersAsync(identifiers, default(CancellationToken))
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Gets the user data corresponding to the specified identifiers.
+        /// <para>
+        /// There are no ordering guarantees; in particular, the nth entry in the users result list
+        /// is not guaranteed to correspond to the nth entry in the input parameters list.
+        /// </para>
+        /// <para>
+        /// A maximum of 100 identifiers may be supplied. If more than 100 identifiers are specified,
+        /// this method throws an <c>ArgumentException</c>.
+        /// </para>
+        /// </summary>
+        /// <param name="identifiers">The identifiers used to indicate which user records should be
+        /// returned. Must have 100 entries or fewer.</param>
+        /// <param name="cancellationToken">A cancellation token to monitor the asynchronous
+        /// operation.</param>
+        /// <returns>A {@code Task} that resolves to the corresponding user records.</returns>
+        /// <exception cref="ArgumentException">If any of the identifiers are invalid or if more
+        /// than 100 identifiers are specified.</exception>
+        public async Task<GetUsersResult> GetUsersAsync(
+            IReadOnlyCollection<UserIdentifier> identifiers, CancellationToken cancellationToken)
+        {
+            var userManager = this.IfNotDeleted(() => this.userManager.Value);
+
+            GetAccountInfoResponse response = await userManager.GetAccountInfoByIdentifiersAsync(identifiers, cancellationToken)
+                .ConfigureAwait(false);
+
+            IEnumerable<UserRecord> userRecords;
```

Comment:
Move all this logic into the constructor of `GetUsersResult`?

Reply:
Done. Also simplified (via linq) similar to your other suggestion.